crates/wasmtime/src/runtime/component/func/typed.rs

```diff
@@ -914,6 +915,44 @@ macro_rules! floats {
                 *ptr = self.to_bits().to_le_bytes();
                 Ok(())
             }
+
+            fn store_list<T>(
+                cx: &mut LowerContext<'_, T>,
+                ty: InterfaceType,
+                offset: usize,
+                items: &[Self],
+            ) -> Result<()> {
+                debug_assert!(matches!(ty, InterfaceType::$ty));
+
+                // Double-check that the CM alignment is at least the host's
+                // alignment for this type which should be true for all
+                // platforms.
+                assert!((Self::ALIGN32 as usize) >= mem::align_of::<Self>());
+
+                // Slice `cx`'s memory to the window that we'll be modifying.
+                // This should all have already been verified in terms of
+                // alignment and sizing meaning that these assertions here are
+                // not truly necessary but are instead double-checks.
+                //
+                // Note that we're casting a `[u8]` slice to `[$integer]` (with
```

Comment:
This paragraph (and the code that follows) refers to `$integer`, but we really care about `$float`, right?  We should rewrite this to talk about floating point numbers instead of integers, e.g. "all `u8` patterns are valid `$float` patterns since `$float` is a IEEE 754 floating point type."